shell/platform/android/io/flutter/plugin/editing/ListenableEditingState.java

```diff
@@ -141,7 +141,11 @@ public void setComposingRange(int composingStart, int composingEnd) {
     if (composingStart < 0 || composingStart >= composingEnd) {
       BaseInputConnection.removeComposingSpans(this);
     } else {
-      mDummyConnection.setComposingRegion(composingStart, composingEnd);
+      if (mDummyConnection != null) {
```

Comment:
Does it make more sense to move the `mDummyConnection` initialization to before the `setEditingState` call in the constructor?

Reply:
I am not sure. Notice mDummyConnection is init as:

```
    Editable self = this;
    mDummyConnection =
        new BaseInputConnection(view, true) {
          @Override
          public Editable getEditable() {
            return self;
          }
        };
```

Where there is a `self` (indeed `this`). If we move it at the beginning of constructor, that `getEditable` can return a not-fully-initialized instance imho

Reply:
If I recall correctly the dummy connection is a quirk to get access to the `BaseInputConnection.setComposingRegion` static method: 
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/inputmethod/BaseInputConnection.java;l=104-128;drc=master?q=BaseInputConnection

Since it's called after the super implementation, and all the dummy connection does is changing the current composing region, it should be fine to return the `Editable` I think (feel free to add a code comment there).

Reply:
fixed